### DISCUSSION TITLE
Adding hatches, linetypes default styles in bonsai

### DISCUSSION THREAD

**Discussion by Slackking1** [IfcOpenShell/IfcOpenShell#5425] [General]
Hi
Hope this is the right place to put this. 
I'm slowly getting into Bonsai, must say i love it. I've been looking for good resources for templates/standard setups. Lots of good stuff already. 
I'm not going to dive into the hardcore dev. of the platform, but i have some .css and .svg experience, would it be of interest that more styles, hatches linetypes etc. got added into the blenderbim/bonsai repo?
I've started making my own, since i do it based on the Danish standards (Molio, which is couple to building smart Denmark) i thought might as well share it and have it built in. 

Was thinking maybe making duplicates of existing files and naming them xxx_DK.xxx, or maybe even just add it to the materials, eg. concrete and concrete_DK, since you then could start using different standards based on regions when doing documentation. In the long run one could imagine this being a setting in the UI, or the plugin setup. 

Not set on how to do it, and if the idea is welcome, I'm open to inputs how to do it to fit into the longer plans of the project.

But, i don't know anything about the underlying projekt yet, which is why I'm asking does it make sense to contribute in this way? Else i will just do it anyway and keep it locally.

**Comment by Moult** [comment]
This is a great idea. Creating user content is always a good thing!

**Reply by Slackking1** [reply]
@Moult What would make the most sense for the long term development, when considering maybe having different regions/documentations standards?

- add styles etc. in default.css and just name it accordingly eg concrete_region/country. From what I see a swap between regions would then require a lot of hardcoding eg. Change each individual defined annotation. 
- make new css and svg and name default_DK.css etc. the upside to this is you could swap fast just changing style etc. in the ui. But it would require a strict naming policy for the swaps to work. 

Any thoughts?

**Reply by Moult** [reply]
The second is preferred. There are ISO standards for country codes that can be used to enforce naming. Collecting a good set of styles is a hard first step. Organisation can be figured out as we go along.

**Reply by Slackking1** [reply]
perfect, i will try to look into it.

**Reply by theoryshaw** [reply]
Would think the following would help with this as well...
That is, be able to chain different css files together.  You could have a core css, overridden by a regional css, for example. 
https://github.com/IfcOpenShell/IfcOpenShell/issues/3451

**Reply by Slackking1** [reply]
Thanks, i will keep that in mind, like the idea. Initial test with @import syntax didn't work though.

**Comment by Slackking1** [comment]
So i made some custom styling, in .svg and .css.  Just in my own repo for now with a project i'm working on. 
Im still adjusting a bit.
See here https://github.com/Slackking1/BonsaiDkStyleTemplate/ 

I tried making a fork and adding them into the folder with the others, but the whole IfcOpenShell/.../data folder is git ignored. 

I've made the following styling and sheets: 
Most work done in the default_DK.css and patterns_DK.svg. 
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/drawings/assets/default_DK.css
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/drawings/assets/markers_DK.svg
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/drawings/assets/patterns_DK.svg
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/drawings/assets/shading_styles_DK.json
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/drawings/assets/symbols_DK.svg
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/layouts/titleblocks/A1_landscape_DK.svg
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/layouts/titleblocks/A2_landscape_DK.svg
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/layouts/titleblocks/A3_landscape_DK.svg
- https://github.com/Slackking1/BonsaiDkStyleTemplate/blob/main/layouts/titleblocks/A4_portrait_DK.svg


QUESTION 1: 
- How to proceed with styling?
     - From a UI/UX experience it would be really nice to be able to set this fast.
- Sheets, how to manage metadata? 
     - Revisions, how to manage this?, Revit has quite a nice way of doing it, that is linked to revision clouds. But for now a manual setting, maybe.
     - Other metadata, as you will see my titleblocks are quite data heavy (but all based on standards). All the information is basically drawing information. A drawing name is a combination of drawing specific metadata, this insures even the name says what the drawing is about. Also i added an image(company logo) to the title blocks, this is setup to use and URI to fetch the logo, guess this should be input somehow. Also there is static data like the company name, maybe that could come from the company in the model settings.
![image](https://github.com/user-attachments/assets/d80c5808-64b6-4f23-a2a9-a8830d784fae)

Just to let you know what i imagine as a potential end result: 
- When you start a project you make a configuration that gives you an initial ifc. 
     - Choosing styling type, eg DK/AU/US etc. maybe even with variations if people contribute, like DK_tech, DK_arch etc.
     - Choose types to use in the projekt, different deciplines could have a custom set of predefined types to include. Then the user only have to add or remove, from that initial point. One setting could be to include some sample drawings and sheets. 

This way at least how i see it, as a beginner in this project, you could make more refined predefined types, materials, renderstyles, shadings etc. Also it would remove the need for company templates, and instead push people in the direction of all contributing to the one library, that let's them start their projekt fast and looking like they want. 

Anyway, i'm willing to put in some hours if you @Moult like this, and think it fits the project. i'm not super skilled in python, but i get along.

Please push be in the right direction from here
